File: Tests/Pages/RegisterPage.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using Tests.Data;

namespace Tests.Pages
{
    public class RegisterPage : BasePage
    {
        public RegisterPage(IWebDriver webDriver) : base(webDriver, "Register") { }

        private IWebElement LoginFld => WebDriver.FindElement(By.Id("login"));
        private IWebElement EmailFld => WebDriver.FindElement(By.Id("email"));
        private IWebElement PasswordFld => WebDriver.FindElement(By.Id("password1"));
        private IWebElement ConfirmFld => WebDriver.FindElement(By.Id("password2"));
        private IWebElement ErrorMsg => WebDriver.FindElement(By.Id("errorMessage"));
        private IWebElement RegisterBtn => WebDriver.FindElement(By.Id("register"));

        public void Register(string login, string email, string password, string confirm = null)
        {
            LoginFld.SendKeys(login);
            EmailFld.SendKeys(email);
            PasswordFld.SendKeys(password);
            ConfirmFld.SendKeys(confirm ?? password);
            RegisterBtn.Click();

            try
            {
                new WebDriverWait(WebDriver, TimeSpan.FromSeconds(Defaults.ImplicitWait))
                .Until(_ => Alert != null || !string.IsNullOrEmpty(Error));
            }
            catch { /**/ }
        }

        public string Error => ErrorMsg?.Text;

        public string Message
        {
            get
            {
                var result = Alert.Text;
                Alert.Accept();

                return result;
            }
        }
    }
}

File: Tests/src/test/java/pages/RegisterPage.java
package test.java.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class RegisterPage extends BasePage {
	@FindBy(id = "login")
	private WebElement login;

	@FindBy(id = "email")
	private WebElement email;

	@FindBy(id = "password1")
	private WebElement password1;

	@FindBy(id = "password2")
	private WebElement password2;

	@FindBy(id = "errorMessage")
	private WebElement errorMessage;

	@FindBy(id = "register")
	private WebElement register;

	public RegisterPage(WebDriver driver) {
		super(driver);
		PageFactory.initElements(this.driver, this);
	}
	
	public String tryRegister(String login, String email, String password1, String password2) {
		this.login.sendKeys(login);
		this.email.sendKeys(email);
		this.password1.sendKeys(password1);
		this.password2.sendKeys(password2);
		this.register.click();
		
		waitForElementVisible(errorMessage);

		return this.errorMessage.getText();
	}
	
	public LoginPage register(String login, String email, String password) {
		this.login.sendKeys(login);
		this.email.sendKeys(email);
		this.password1.sendKeys(password);
		this.password2.sendKeys(password);
		this.register.click();
		
		waitForAlert();
		driver.switchTo().alert().accept();
		
		return new LoginPage(driver);
	}
}